cql3/statements/alter_keyspace_statement.cc

```diff
@@ -81,33 +86,80 @@ void cql3::statements::alter_keyspace_statement::validate(query_processor& qp, c
 #endif
 }
 
+static logging::logger mylogger("alter_keyspace");
+
 future<std::tuple<::shared_ptr<cql_transport::event::schema_change>, std::vector<mutation>, cql3::cql_warnings_vec>>
 cql3::statements::alter_keyspace_statement::prepare_schema_mutations(query_processor& qp, const query_options&, api::timestamp_type ts) const {
+    using namespace cql_transport;
     try {
-        auto old_ksm = qp.db().find_keyspace(_name).metadata();
+        event::schema_change::target_type target_type = event::schema_change::target_type::KEYSPACE;
+        auto ks = qp.db().find_keyspace(_name);
+        auto ks_md = ks.metadata();
+
         const auto& tm = *qp.proxy().get_token_metadata_ptr();
         const auto& feat = qp.proxy().features();
+        auto ks_md_update = _attrs->as_ks_metadata_update(ks_md, tm, feat);
+        std::vector<mutation> muts;
+        std::vector<sstring> warnings;
+
+        if (ks.get_replication_strategy().uses_tablets() && !rf_map.empty()) {
+            qp.db().real_database().validate_keyspace_update(*ks_md_update);
+            if (auto repl_strategy = _attrs->get_replication_strategy_class();
+                    repl_strategy.has_value() && repl_strategy != "NetworkTopologyStrategy")
+                throw make_exception_future<std::tuple<::shared_ptr<::cql_transport::event::schema_change>, std::vector<mutation>, cql3::cql_warnings_vec>>(
+                        exceptions::invalid_request_exception("For tablets-enabled keyspaces, only NetworkTopologyStrategy replication strategy is supported"));
+
+            if (qp.topology_global_queue_empty()) {
+                service::topology_mutation_builder builder(ts);
+                builder.set_global_topology_request(service::global_topology_request::keyspace_rf_change);
+                builder.set_global_topology_request_id(this->global_req_id);
+                // TODO: need to correctly handle the generic 'replication_factor' tag in NTS
+                //       currently, if it's used, we're passing an empty map and some test.py testcases fail
+                builder.set_new_keyspace_rf_change_data(_name, _attrs->get_replication_map());
+                service::topology_change change{{builder.build()}};
+                auto topo_schema = qp.db().find_schema(db::system_keyspace::NAME, db::system_keyspace::TOPOLOGY);
+                boost::transform(change.mutations, std::back_inserter(muts), [topo_schema] (const canonical_mutation& cm) {
+                    return cm.to_mutation(topo_schema);
+                });
```

Comment:
btw, we transform those canonical mutations back and forth into mutations (for `redact_columns_for_missing_features`), I wonder if this could be improved in the future (out of scope for this PR of course)

Reply:
Yes, I have to translate them to whatever format the consumer (API) expects them in. Fortunately, there are only 2 formats - not that bad.